src/mantarray_waveform_analysis/pipelines.py

```diff
@@ -241,6 +344,40 @@ def get_compressed_displacement(self) -> NDArray[(2, Any), np.float32]:
         )
         return self._compressed_displacement
 
+    def get_compressed_force(self) -> NDArray[(2, Any), np.float32]:
+        try:
+            return self._compressed_force
+        except AttributeError:
+            pass
+        self._compressed_force = calculate_force_from_displacement(
+            self.get_compressed_displacement()
+        )
+        return self._compressed_force
+
+    def get_voltage(self) -> NDArray[(2, Any), np.float32]:
+        try:
+            return self._voltage
+        except AttributeError:
+            pass
+        self._voltage = calculate_voltage_from_gmr(self.get_noise_filtered_gmr())
```

Comment:
note for future: we're not actually using GMR sensors anymore. so maybe something more generic like `get_noise_filtered_sensor_reading` (and generally replacing gmr with `sensor_reading` or something similar)